presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestAggregations.java

```diff
@@ -660,12 +660,51 @@ public void test15WayGroupBy()
     @Test
     public void testApproximateCountDistinct()
     {
+        // test date
+        assertQuery("SELECT approx_distinct(orderdate) FROM orders", "SELECT 2443");
+        assertQuery("SELECT approx_distinct(orderdate, 0.023) FROM orders", "SELECT 2443");
+
+        // test timestamp
```

Comment:
what about TIME/TIME WTZ/TIMESTAMP WTZ?

Reply:
I figured out there is a convenient magic function `$literal` that can be used for test. I will add all missing types in this smoke test.

Reply:
The purpose of the magic literal is the "transfer" of the values that have no literal. Although it is possible to call it explicitly now, it might not be in the future.